src/CTA.WebForms2Blazor/Services/TaskManagerService.cs

```diff
@@ -36,18 +30,25 @@ public int RegisterNewTask()
 
             return newManagedTask.TaskId;
         }
-        
-        public CancellationToken SetTaskWaiting(int taskId, TaskStallTimeout stallTimeout = TaskStallTimeout.Short)
-        {
-            var token = _managedTasks[taskId].SetWaiting(stallTimeout);
-            UpdateStallState();
-            return token;
-        }
 
-        public void SetTaskActive(int taskId)
+        public async Task<TResult> ManagedRun<TResult>(int taskId, Func<CancellationToken, Task<TResult>> func)
         {
-            _managedTasks[taskId].SetActive();
+            TResult result;
+            var managedTask = _managedTasks[taskId];
+            var token = managedTask.SetWaiting();
             UpdateStallState();
+
+            try
+            {
+                result = await func(token);
+                managedTask.SetActive();
+            }
+            finally
+            {
+                UpdateStallState();
+            }
+
+            return result;
         }
```

Comment:
This is my understanding as well

Reply:
This is partially correct, the func uses the cancellation token before it become null, because the task is only set to active once the awaited func returns, either via an exception (from cancellation) or success, after which the token is no longer needed. However, yes with a single task it will immediately cancel the operation, this is intentional and I believe is what we discussed the functionality would be until we decide to uncomment the code with the delay in it

Reply:
If we want, I could also make it so that if only 1 task exists it will not count it as a stall

Reply:
When a task is cancelled, the task state is set to `active`. In my hypothetical case, this would occur in the first call to UpdateStallState(), before the `func` is executed (call chain would look like: `UpdateStallState()` -> `TryResolveStall()` -> `CancelTask()`).

---

I don't think we should specifically target the case where 1 task exists, my example was only to serve as a base case. My thinking is that if there's concern with the case where only 1 task exists, then we should be concerned if multiple managed tasks run concurrently. 

The reason is that if managed tasks are running concurrently, then it's possible for all tasks to be in a state where they're set to `waiting`, leading the app to believe it's in a stalled state when it really just hasn't begun processing anything yet. However unlikely this case seems, especially if we're anticipating many managed tasks, it's an edge case we need to account for (production environments love Murphy's Law 🙂 ). If encountered in production, it would be incredibly difficult to debug.

---

Follow-up: if you believe this is a concern, I have a few ideas for resolution but they might need review:
1. Add additional state variable specifying whether or not the first task has officially attempted to run
2. Update TaskManager to function like a state machine with states like: `init`, `running`, `stalled`, `complete` (or whatever makes sense) and have explicit functions and conditions that transition state (we sort of have that in place now, but this pattern might make the code easier to test and maintain)

Reply:
Ah ok, I see what you're saying now. Yes, I forgot that cancel sets the state to active, it still functions not using a null token because the token is retrieved when the wait state is set, before the update stall state call occurs. So the token, when passed to func, will already be set to cancelled and token source on the managed task object will be null. I think you're right that some more advanced state management could be useful here, I'll make the change

Reply:
Per our conversation, this is no longer a concern. Should be fine as-is